src/parser/priest/discipline/modules/azeritetraits/EnduringLuminescence.js

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import { calculateTraitHealing } from 'parser/shared/modules/helpers/CalculateTraitHealing';
+import { calculateAzeriteEffects } from 'common/stats';
+import { formatPercentage } from 'common/format';
+import AzeritePowerStatistic from 'interface/statistics/AzeritePowerStatistic';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import ItemHealingDone from 'interface/ItemHealingDone';
+import Analyzer from 'parser/core/Analyzer';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+import isAtonement from '../core/isAtonement';
+
+import { POWER_WORD_RADIANCE_COEFFICIENT } from '../../constants';
+
+
+const ENDURING_LUMINESCENCE_BONUS_MS = 1500;
+const EVANGELISM_BONUS_MS = 6000;
+
+/*
+* Power Word: Radiance restores 1672 additional health, and applies Atonement for 70% of its normal duration.
+*/
+class EnduringLuminescense extends Analyzer {
+    static dependencies = {
+        statTracker: StatTracker,
+    }
+    _bonusFromAtonementDuration = 0;
+    _bonusFromDirectHeal = 0;
+
+    _rawBonusHealingPerHit = 0;
+
+    _lastCastIsPowerWordRadiance = false;
+    _atonementsAppliedByRadiances = [];
+
+    constructor(...args) {
+        super(...args);
+        this._maxTargets = 5;
+        this.active = this.selectedCombatant.hasTrait(SPELLS.ENDURING_LUMINESCENCE.id);
+
+        if (!this.active) {
```

Comment:
Point less code if you don't use actionlisteners. 
Please update to use actions listeners